trl/trainer/grpo_trainer.py

```diff
@@ -820,6 +820,8 @@ def __init__(
                     max_num_batched_tokens=4096,
                     model_impl=self.args.vllm_model_impl,
                 )
+            else:
```

Comment:
I don't think we need a check in `grpo_trainer`. If no config object is passed, this check will never occur because `self.use_vllm`  will be false since the Config object sets this to false by default.

https://github.com/huggingface/trl/blob/2fb70902313c907a06f55aeeab9f81ce72caaff5/trl/trainer/grpo_trainer.py#L509-L512

If the use passes a config object, the check in the config file will raise the exception.

Reply:
Ahhh I see, it'd be quite strange for someone to do that but I don't have a strong opinion, so feel free to leave that check in.

Reply:
Updated!